src/FsMutationsDataReader.cpp

```diff
@@ -92,42 +112,100 @@ void FsMutationsDataReader::createJSON(Fmq* pending, INodeMap& inodes,
         LOG_DEBUG(" Data for xattr: " << row.getXAttrName() << ", "
         << row.getNamespace() <<  " for inode " << row.mInodeId
         << " was not found");
-        bulk.push(mFSLogTable.getLogRemovalHandler(row), row
-            .mEventCreationTime, XAttrRow::to_delete_json(row));
+        if(FileProvenanceConstants::isFeaturegroup(inode.mId, datasetINodeId, projectName, datasetName)) {
+          bulk.push(nullptr, row.mEventCreationTime, XAttrRow::to_delete_json(mFeaturestoreIndex, row));
+        }
+        bulk.push(mFSLogTable.getLogRemovalHandler(row), row.mEventCreationTime, XAttrRow::to_delete_json(mSearchIndex, row));
         continue;
       }
 
       XAttrVec xattr = xattrs[mutationpk];
       if(xattr.empty()){
         LOG_DEBUG(" Data for all xattrs of inode " << row.mInodeId
                                       << " was not found");
-        bulk.push(mFSLogTable.getLogRemovalHandler(row), row
-            .mEventCreationTime, XAttrRow::to_delete_json(row));
+        if(FileProvenanceConstants::isFeaturegroup(inode.mId, datasetINodeId, projectName, datasetName)) {
+          bulk.push(nullptr, row.mEventCreationTime, XAttrRow::to_delete_json(mFeaturestoreIndex, row));
+        }
+        bulk.push(mFSLogTable.getLogRemovalHandler(row), row.mEventCreationTime, XAttrRow::to_delete_json(mSearchIndex, row));
         continue;
       }
 
       int i = 0;
       for(XAttrVec::iterator it = xattr.begin(); it != xattr.end() ; ++it, i++){
         //only add the first event with a removal handler
-        const LogHandler* const logh = i > 0 ? nullptr : mFSLogTable
-            .getLogRemovalHandler(row);
+        const LogHandler *const logh = i > 0 ? nullptr : mFSLogTable.getLogRemovalHandler(row);
```

Comment:
i thought you are going to change this to have log handler only at the last event?

Reply:
forgot about this one... fixed